OrbitQt/ConnectToStadiaWidget.cpp

```diff
@@ -0,0 +1,378 @@
+// Copyright (c) 2020 The Orbit Authors. All rights reserved.
+// Use of this source code is governed by a BSD-style license that can be
+// found in the LICENSE file.
+
+#include "ConnectToStadiaWidget.h"
+
+#include <QMessageBox>
+#include <QRadioButton>
+#include <QSettings>
+#include <optional>
+#include <utility>
+
+#include "Error.h"
+#include "OrbitBase/Logging.h"
+#include "OrbitBase/Result.h"
+#include "OrbitGgp/Client.h"
+#include "OverlayWidget.h"
+#include "servicedeploymanager.h"
+
+namespace {
+const QString kRememberChosenInstance{"RememberChosenInstance"};
+}  // namespace
+
+namespace orbit_qt {
+
+using OrbitGgp::Instance;
+using OrbitSshQt::ScopedConnection;
+
+ConnectToStadiaWidget::ConnectToStadiaWidget(QWidget* parent)
+    : QWidget(parent), ui_(std::make_unique<Ui::ConnectToStadiaWidget>()) {
+  ui_->setupUi(this);
+
+  parent->installEventFilter(this);
+
+  QSettings settings;
+  if (settings.contains(kRememberChosenInstance)) {
+    remembered_instance_id_ = settings.value(kRememberChosenInstance).toString();
+    ui_->rememberCheckBox->setChecked(true);
+  }
+
+  ui_->instancesTableView->setModel(&instance_model_);
+
+  QObject::connect(ui_->connectToStadiaInstanceRadioButton, &QRadioButton::clicked, this,
+                   &ConnectToStadiaWidget::OnConnectToStadiaRadioButtonClicked);
+  QObject::connect(this, &ConnectToStadiaWidget::ErrorOccurred, this,
+                   &ConnectToStadiaWidget::OnErrorOccurred);
+  QObject::connect(ui_->instancesTableView->selectionModel(), &QItemSelectionModel::currentChanged,
+                   this, &ConnectToStadiaWidget::OnSelectionChanged);
+  QObject::connect(ui_->rememberCheckBox, &QCheckBox::toggled, this,
+                   &ConnectToStadiaWidget::OnRememberCheckBoxToggled);
+
+  SetupAndStartStateMachine();
+}
+
+bool ConnectToStadiaWidget::eventFilter(QObject* obj, QEvent* event) {
+  if (obj == parent() && event->type() == QEvent::Resize) {
+    ui_->contentFrame->resize(size());
+  }
+  return false;
```

Comment:
According to the docs  you're supposed to call QObject::eventFilter when your custom logic doesn't match.
Not sure if that's just good practice or actually needed.
Details: https://doc.qt.io/qt-5/qobject.html#installEventFilter

Reply:
Ive also read these docs and I feel they are on the rather ambiguous side. The text says to either return true in the case it should not be propagated further (filtered) and return false if the event should not be filtered. 
The example then does not return true or false, but propagates the event manually by calling QObject::eventFilter with the obj as parameter. 
I am unsure what to make of this. It seems to me that the example is rather wrong and might might not work when multiple event filter are installed.

Reply:
I don't see how this is related to multiple event filters?

I think this example is clearer: https://doc.qt.io/qt-5/qobject.html#eventFilter
In that example you have all three cases. Explicit true, explicit false and forwarding to base class implementation.

I'd argue since we don't want to change any behaviour we should return the value that would be returned without overriding the member function at all. And that's the return value of the base class implementation.

Reply:
Yes, I of course dont want to change any behaviour. I did not fully understand before how this actually works. But I think I now got it. The return true/false is about whether another event filter, maybe on the parent, or any other installed eventfilter should be called. The direction up the inheritance chain (base class) is independent of that.

Reply:
To my understanding, the true / false return value is not primarily for other filters. The main purpose of the eventFilter is not to handle events, but to prevent them from being handled by the object. Event filters are called in the order they are installed, and I think if any of them returns `true`, the event is not passed to the target object.

Installing event filters to react to events of other widgets is, IMHO, abusing them a bit. But it's also a quick and easy way to achieve what you are trying to do, otherwise you'd either have to fiddle around with custom layouts, or provide a `Resized()` signal in the parent and connect this to a slot of the child.

Reply:
I found a way to use a regular layout and not use eventFilter at all. 
Thank you both anyways for taking the time to explain